### PR TITLE
Feat/desk booking refresh

### LINKED ISSUE #168: WPA Desk Booking Flow Tests

### PR DESCRIPTION
**Pull request by vinnyo**
**Description of the change**

Various fixes and tests for new desk booking flow

### CODE REVIEW

File: apps/workplace/src/tests/book/new-desk-flow.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            spectator.detectChanges();
+        });
+        book_service.setView('form');
+    });
+
+    it('should create component', () =>
+    expect(spectator.component).toBeTruthy());
+
+    it('should show form view by default', () => {
+        expect(spectator.query('new-desk-flow-form')).toExist();
```
Comment (MrYuion) [inline apps/workplace/src/tests/book/new-desk-flow.component.spec.ts]: Note that you don't need to do `spectator.query` for `.toExist` checks.
`expect('new-desk-flow-form').toExist();` works the same and is less verbose.

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-confirm.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+describe('NewDeskFlowConfirmComponent', () => {
+    let spectator: Spectator<NewDeskFlowConfirmComponent>;
+    const createComponent = createRoutingFactory({
+        component: NewDeskFlowConfirmComponent,
+        providers: [
+            {
+                provide: BookingFormService,
+                useValue: {
+                    form: new FormGroup({
+                        date: new FormControl(new Date()),
```
Comment (MrYuion) [inline apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-confirm.component.spec.ts]: I feel like this form group is mostly implementation details and isn't necessary for the tests.
If you do need the correct form use the function from the utilities to generate it.

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-success.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { MatDialog } from '@angular/material/dialog';
+import { createRoutingFactory, Spectator } from '@ngneat/spectator/jest';
+import { NewDeskFlowSuccessComponent } from 'apps/workplace/src/app/book/new-desk-flow/new-desk-flow-success.component';
+
+describe('NewDeskFlowSuccessComponent', () => {
+    let spectator: Spectator<NewDeskFlowSuccessComponent>;
+    const createComponent = createRoutingFactory({
```
Comment (MrYuion) [inline apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-success.component.spec.ts:7]: Minor note but I routing isn't needed here so use `createComponentFactory` instead.
In my snippets in VSCode for generating the boilerplate for tests I mistakenly used the `createRoutingFactory` instead `createComponentFactory`. Sorry for adding a bit of extra work on your end.

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-form.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import { BookingFormService } from '@placeos/bookings';
+import { Building, OrganisationService } from '@placeos/organisation';
+import { NewDeskFlowFormComponent } from 'apps/workplace/src/app/book/new-desk-flow/new-desk-flow-form.component';
+import { NewDeskFormDetailsComponent } from 'apps/workplace/src/app/book/new-desk-flow/new-desk-form-details.component';
+import { MockComponent } from 'ng-mocks';
+import { BehaviorSubject, of } from 'rxjs';
+
+describe('NewDeskFlowFormComponent', () => {
+    let spectator: Spectator<NewDeskFlowFormComponent>;
+    const createComponent = createRoutingFactory({
```
Comment (MrYuion) [inline apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-form.component.spec.ts:14]: Minor note but I routing isn't needed here so use `createComponentFactory` instead.
In my snippets in VSCode for generating the boilerplate for tests I mistakenly used the `createRoutingFactory` instead `createComponentFactory`. Sorry for adding a bit of extra work on your end.

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-success.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        component: NewDeskFlowSuccessComponent,
+        providers: [{ provide: MatDialog, useValue: { open: jest.fn() } }],
+        declarations: [],
+    });
+
+    beforeEach(() => (spectator = createComponent()));
+
+    it('should create component', () =>
+        expect(spectator.component).toBeTruthy());
```
Comment (MrYuion) [inline apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-success.component.spec.ts:25]: Is there any behaviour in this component that needs testing?

File: libs/bookings/src/lib/booking.utilities.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ export function generateBookingForm(booking: Booking = new Booking()) {
         title: new FormControl(booking.title),
         description: new FormControl(booking.description),
         booking_asset: new FormControl(null),
-        asset_id: new FormControl(booking.asset_id, [Validators.required]),
+        asset_id: new FormControl(booking.asset_id),
```
Comment (MrYuion) [inline libs/bookings/src/lib/booking.utilities.ts]: Can you undo this as this should be a required field.
If you are doing full validity checks it should be at the last step in the flow.

**Review by MrYuion — CHANGES_REQUESTED** [review]
Looks good. Just some minor things I've commented on.

File: libs/bookings/src/lib/booking.utilities.ts (hunk 6 above)
Comment (vinnyo) [reply libs/bookings/src/lib/booking.utilities.ts]: Hmm, I previously added this for the new flow, but found that it made the old flow stop working. Do we still want to maintain the old flow?

File: apps/workplace/src/tests/book/new-desk-flow.component.spec.ts (hunk 1 above)
Comment (vinnyo) [reply apps/workplace/src/tests/book/new-desk-flow.component.spec.ts]: Fixed accordingly

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-confirm.component.spec.ts (hunk 2 above)
Comment (vinnyo) [reply apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-confirm.component.spec.ts]: Using generate form from utilities

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-form.component.spec.ts (hunk 4 above)
Comment (vinnyo) [reply apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-form.component.spec.ts:14]: Resolved as discussed

File: apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-success.component.spec.ts (hunk 5 above)
Comment (vinnyo) [reply apps/workplace/src/tests/book/new-desk-flow/new-desk-flow-success.component.spec.ts:25]: Add some test for success flow view

File: libs/bookings/src/lib/booking.utilities.ts (hunk 6 above)
Comment (vinnyo) [reply libs/bookings/src/lib/booking.utilities.ts]: Undo changes as discussed

**Review by MrYuion — APPROVED** [review]
Looks good.

**Pull request merged**